### ISSUE TITLE
Sonos: Event published as "Undefined" 

### ISSUE THREAD

**Issue by nanosonde** [eclipse/smarthome#3491]
I was sent over from [this](https://github.com/openhab/openhab1-addons/issues/4902) issue concerning the Sonos binding.

The Sonos binding is causing an "Undefined" event on the MQTT eventbus topic when playing or pausing a Sonos speaker.

After some investigating I found out that this is possibly related to state transition "TRANSITIONING" which is also sent by the player besides the states "PLAYING" and "PAUSING".

Please read my [comment ](https://github.com/openhab/openhab1-addons/issues/4902#issuecomment-301108180)in the other issue.

**Comment by lolodomo** [comment]
Yes the transitioning state exists since ever and is only used by Sonos when going from pause/stop to playing state.
When this state is used by Sonos, thee binding is setting the channel to UNDEF but there is always another final state dent by Sonos that will lead to the channel being set to a non UNDEF state.
All this looks good to me.

**Comment by kaikreuzer** [comment]
"All this looks good to me."

@lolodomo You mean the current behavior looks good to you and thus the issue should be closed?

**Comment by lolodomo** [comment]
Yes but that is only my opinion.

**Comment by kaikreuzer** [comment]
Do you see any value in having the Undefined intermediate state? If not, I would agree with @nanosonde that it should not be set/send, but rather be swallowed.

**Comment by lolodomo** [comment]
Transitioning state is considered as Undefined by ESH.
If you prefer ignoring this state, the binding code has to be changed.

**Comment by nanosonde** [comment]
Hi!
Ok, maybe the problem is only linked to the case where both bindings (sonos and MQTT-eventbus) should work together.

The original problem that was reported in [this](https://github.com/openhab/openhab1-addons/issues/4902) issue is that only the aforementioned UNDEF state can be seen with MQTT on the event bus.
The final states (PLAYING/STOPPED) are never transmitted over MQTT.

I was assuming that this is somehow linked to this transitional state which gets published only instead of one of the final state.

**Comment by lolodomo** [comment]
There is certainly another reason. Check your event logs and you will see that the PLAYING state is sent.

**Comment by lolodomo** [comment]
You can also check in any UI that the player state is defined.

**Comment by nanosonde** [comment]
Yes, the player state is defined in the UI and the event log also shows the correct final state.
But, I have no explanation why the MQTT eventbus binding behaves this way when used with the Sonos binding.

I have setup the OH dev env to debug the problem. So far I can only observe this:
* The MQTT eventbus binding gets events from the event bus.
* The MQTT eventbus binding ONLY sees the UNDEF state for Sonos. No other state.
* The MQTT eventbus binding ONLY sees exactly ONE state update for Sonos, regardless of PLAYING->STOPPED/PAUSING or STOPPED/PAUSING->PLAYING: Here I would expect at least two events for the case with the TRANSITIONAL state.

**Comment by lolodomo** [comment]
It looks like a bug in the MQTT event bus. Maybe the player state is not considered by MQTT  ?

**Comment by lolodomo** [comment]
I finally encountered a case where channel updates are missing leading to the channel being set only to TRANSITIONING : issue #4447 
Unfortunately I am not able to reproduce it on my device environment.

Are you running OH on a slow machine like a RPI 2 ?

**Comment by bummzulu** [comment]
I am facint the same issue. When using the Sonos App to control a speaker the state of "control" does often not change from Pause to Play but rather Pause to Undef to Play. This is causing an issue with my openhab rules configuration.

I am running openhab on a RPI3.
